src/Compilers/CSharp/Portable/Symbols/AnonymousTypes/AnonymousTypeDescriptor.cs

```diff
@@ -111,14 +109,8 @@ internal AnonymousTypeDescriptor WithNewFieldsTypes(ImmutableArray<TypeSymbolWit
             Debug.Assert(!newFieldTypes.IsDefault);
             Debug.Assert(newFieldTypes.Length == this.Fields.Length);
 
-            AnonymousTypeField[] newFields = new AnonymousTypeField[this.Fields.Length];
-            for (int i = 0; i < newFields.Length; i++)
-            {
-                var field = this.Fields[i];
-                newFields[i] = new AnonymousTypeField(field.Name, field.Location, newFieldTypes[i]);
-            }
-
-            return new AnonymousTypeDescriptor(newFields.AsImmutable(), this.Location);
+            var newFields = this.Fields.SelectAsArray((field, i, types) => new AnonymousTypeField(field.Name, field.Location, types[i]), newFieldTypes);
```

Comment:
This is a suble reminder I need to push for C#8 being allowed in the repo. Then we could have named this newFieldTypes as well.